sql/core/src/main/scala/org/apache/spark/sql/execution/Expand.scala

```diff
@@ -71,9 +80,86 @@ case class Expand(
             idx = 0
           }
 
+          numOutputRows += 1
           result
         }
       }
     }
   }
+
+  override def upstream(): RDD[InternalRow] = {
+    child.asInstanceOf[CodegenSupport].upstream()
+  }
+
+  protected override def doProduce(ctx: CodegenContext): String = {
+    child.asInstanceOf[CodegenSupport].produce(ctx, this)
+  }
+
+  override def doConsume(ctx: CodegenContext, input: Seq[ExprCode]): String = {
+    // Some columns have the same expression in all the projections, so collect the unique
+    // expressions.
+    val columnUniqueExpressions: IndexedSeq[Set[Expression]] = output.indices.map { i =>
```

Comment:
for this one, can we explain what the indexes are, and what the expressions are?